ports/unix/unix_mphal.c

```diff
@@ -173,7 +176,9 @@ int mp_hal_stdin_rx_chr(void) {
 }
 
 void mp_hal_stdout_tx_strn(const char *str, size_t len) {
+    MP_THREAD_GIL_EXIT();
     int ret = write(1, str, len);
+    MP_THREAD_GIL_ENTER();
```

Comment:
This may be inefficient when printing a lot of text in small chunks.  On esp32 this was a real problem (REPL was very slow with GIL enabled) so the GIL is released only if printing more than 20 characters (see `ports/esp32/mphalport.c:mp_hal_stdout_tx_strn`).  But on esp32 we know that printing chars won't block for long on the UART so it's fair to do that.  Here on unix we have no guarantees/knowledge about `write` and it may block for a long time on a single char, so I guess the patch here is the best we can do.